### ISSUE TITLE
Failed to migrate legacy database error="FOREIGN KEY constraint failed"

### ISSUE THREAD

**Issue by freaktechnik** [mautrix/whatsapp#725]
Bridge isn't starting, failing with the same error every time.
```
Oct  4, 2024 12:02:49 INF Initializing bridge built_at=2024-10-03T11:39:51Z go_version=go1.23.2 name=mautrix-whatsapp version=0.11.0+dev.a58f6a91
Oct  4, 2024 12:02:49 DBG Initializing database connection
Oct  4, 2024 12:02:51 INF Detected legacy database, migrating... action="migrate legacy db"
Oct  4, 2024 12:02:51 FTL Failed to migrate legacy database error="FOREIGN KEY constraint failed" db_section=main
```

Edit: identical logs with `version=0.11.0+dev.1f454a82`.

**Comment by progecon** [comment]
I am facing the same issue, the mautrix container keeps restarting

**Comment by Glotzon** [comment]
Same isssue with my instance:

1. ``synapse-whatsapp-1  | Oct  6, 2024 08:40:47 INF Initializing bridge built_at=2024-10-04T11:43:44Z go_version=go1.23.2 name=mautrix-whatsapp version=0.11.0+dev.1f454a82
2. synapse-whatsapp-1  | Oct  6, 2024 08:40:47 DBG Initializing database connection
3. synapse-whatsapp-1  | Oct  6, 2024 08:40:47 WRN Using SQLite without _txlock=immediate is not recommended fixed_uri_example=file:mautrix-whatsapp.db?_txlock=immediate
4. synapse-whatsapp-1  | Oct  6, 2024 08:40:47 INF Detected legacy database, migrating... action="migrate legacy db"
5. synapse-whatsapp-1  | Oct  6, 2024 08:40:47 FTL Failed to migrate legacy database error="FOREIGN KEY constraint failed" db_section=main
6. synapse-whatsapp-1 exited with code 15``>

**Comment by freaktechnik** [comment]
Still failing:
```
Oct 16, 2024 14:06:41 INF Initializing bridge built_at=2024-10-16T11:08:42Z go_version=go1.23.2 name=mautrix-whatsapp version=0.11.0
Oct 16, 2024 14:06:41 DBG Initializing database connection
Oct 16, 2024 14:06:41 INF Detected legacy database, migrating... action="migrate legacy db"
Oct 16, 2024 14:06:41 FTL Failed to migrate legacy database error="FOREIGN KEY constraint failed" db_section=main
``

**Comment by freaktechnik** [comment]
This si fixed for me as of 220dc219b8f8cc74c39ff42521cf2675188899c9
